Allow dotCLI to push very large files

### Parent Issue

_No response_

### User Story

As a user, I may need to move files larger than 50 mb from a workstation to an instance. Whereas we're looking for dotCLI to eventually take on all of WebDAV's old duties, that should ultimately include the management of files of arbitrary size.

### Acceptance Criteria

Remove the 50 mb limit on dotCLI file transfers, and allow the command line to move files up to the maximum size the system itself can allow.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

Current errors received when attempting to push large files:

```
[ERROR] ❗  Error pushing asset [/test2g] 
 RESTEASY005081: File limit of 50MB has been reached. The entity cannot be processed. Increase the size with the configuration property dev.resteasy.entity.file.threshold.
[ERROR] ❗  Error pushing asset [/test5g] 
 RESTEASY005081: File limit of 50MB has been reached. The entity cannot be processed. Increase the size with the configuration property dev.resteasy.entity.file.threshold.
[ERROR] ❗  Error pushing asset [/test10g] 
 RESTEASY005081: File limit of 50MB has been reached. The entity cannot be processed. Increase the size with the configuration property dev.resteasy.entity.file.threshold.
```

Quarkus's `dev.resteasy.entity.file.threshold` is at the heart of this, but @fabrizzio-dotCMS and I were not able to find a quick and easy workaround at the level of local variables.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

--- a/tools/dotcms-cli/api-data-model/src/main/resources/application.properties
+++ b/tools/dotcms-cli/api-data-model/src/main/resources/application.properties
@@ -35,3 +35,6 @@ com.dotcms.starter.site=default
 %test.com.dotcms.starter.site=default
 # We can also start our tests using a demo profile to test against a demo site
 %demo.com.dotcms.starter.site=demo.dotcms.com
+
+# The threshold to use for the amount of data that can be stored in a file for entities. -1 means no limit is set.
+dev.resteasy.entity.file.threshold=-1

--- a/tools/dotcms-cli/cli/src/test/java/com/dotcms/api/client/files/PushServiceIT.java
+++ b/tools/dotcms-cli/cli/src/test/java/com/dotcms/api/client/files/PushServiceIT.java
@@ -41,6 +41,8 @@ import java.util.UUID;
 import java.util.stream.Stream;
 import jakarta.inject.Inject;
 import org.apache.commons.io.FileUtils;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Disabled;
@@ -949,6 +951,21 @@ class PushServiceIT {
         final var remoteAssetPath = buildRemoteAssetURL(siteName, folderPath, assetName);
         Assertions.assertTrue(filesTestHelper.assetExist(remoteAssetPath),
                 String.format("Asset %s was not created", remoteAssetPath));
+    }
+
+
+    /**
+     * Verifies that the RESTEasy file upload threshold is properly configured.
+     * This test ensures that the 'quarkus.resteasy.multipart.file-size-threshold' property 
+     * is set to -1, which allows for unlimited file sizes during multipart file uploads.
+     * Setting this value to -1 is crucial for handling large file uploads
+     */
+    @Test
+    void testFileThresholdProperty() {
+        final Config config = ConfigProvider.getConfig();
+        final Long fileThreshold = config.getValue("dev.resteasy.entity.file.threshold", Long.class);
+        // Assert that the property is correctly set to -1
+        Assertions.assertEquals(-1L, fileThreshold, "The file threshold should be set to -1 for unlimited file size.");
     }
 
 }

